mlfinlab/cross_validation/cross_validation.py

```diff
@@ -0,0 +1,114 @@
+"""
+Implements the book chapter 7 on Cross Validation for financial data
+"""
+
+import pandas as pd
+import numpy as np
+
+from sklearn.metrics import log_loss, accuracy_score
+from sklearn.model_selection import KFold
+
+
+def ml_get_train_times(info_sets: pd.Series, test_times: pd.Series) -> pd.Series:
+    # pylint: disable=invalid-name
+    """
+    Snippet 7.1, page 106,  Purging observations in the training set
+
+    This function find the training set indexes given the information on which each record is based
+    and the range for the test set.
+
+    Given test_times, find the times of the training observations.
+    :param info_sets: The information on which each record is constructed from
+        -info_sets.index: Time when the information extraction started.
+        -info_sets.value: Time when the information extraction ended.
+    :param test_times: Times for the test dataset.
+    """
+    train = info_sets.copy(deep=True)
+    for start_ix, end_ix in test_times.iteritems():
+        df0 = train[(start_ix <= train.index) & (train.index <= end_ix)].index  # Train starts within test
+        df1 = train[(start_ix <= train) & (train <= end_ix)].index  # Train ends within test
+        df2 = train[(train.index <= start_ix) & (end_ix <= train)].index  # Train envelops test
+        train = train.drop(df0.union(df1).union(df2))
+    return train
+
+
+class PurgedKFold(KFold):
+    """
+    Extend KFold class to work with labels that span intervals
+    The train is purged of observations overlapping test-label intervals
+    Test set is assumed contiguous (shuffle=False), w/o training samples in between
+    """
+
+    def __init__(self, n_splits=3, info_sets=None, pct_embargo=0.):
+        """
+        :param n_splits: The number of splits. Default to 3
+        :param info_sets:
+            —info_sets.index: Time when the information extraction started.
+            —info_sets.value: Time when the information extraction ended.
+        :param pct_embargo: Percent that determines the embargo size.
+        """
+        if not isinstance(info_sets, pd.Series):
+            raise ValueError('The info_sets param must be a pd.Series')
+        super(PurgedKFold, self).__init__(n_splits, shuffle=False, random_state=None)
+
+        self.info_sets = info_sets
+        self.pct_embargo = pct_embargo
+
+    # noinspection PyPep8Naming
+    def split(self, X, y=None, groups=None):
+        if len(X.index) != len(self.info_sets.index) or \
```

Comment:
Hello
This will raise error when using sklearn
If `rfecv = sklearn.feature_selection.RFECV(..., cv=PurgedKFold(...), ...)`
then `rfecv.fit(X, y)` will raise error because sklearn replace X with X.values (a np.ndarray) in `rfecv.fit()`
Maybe treating X as a np.ndarray would be better?

Reply:
Good point @lunluen!